xerrors: improve error message on parsing metasheet

As metasheet `@TABLEAU` is defined as:
| Sheet| OrderedMap | Mode |
|--------|--------|------|
| Item |  |  |
| Enum | MODE_ENUM_TYPE_MULTI |  |

An error occured on parsing:
```
error[E2013]: invalid syntax of boolean value
工作簿: conf/Item.xlsx: failed to read book: conf/Item.xlsx: failed to parse metasheet: @TABLEAU: failed to parse metasheet 
工作表: @TABLEAU 
命名单元格位置: <no value>
命名单元格数据: <no value>
类型单元格位置: <no value>
类型单元格数据: <no value>|name:"OrderedMap" prop:{optional:true}
错误原因: 无法将 "MODE_ENUM_TYPE_MULTI" 解析为布尔类型, strconv.ParseBool: parsing "MODE_ENUM_TYPE_MULTI": invalid syntax
```

The problem is:
- error message has no cell position and cell data filled
- concrete error message mixes in workbook

--- a/internal/confgen/document_parser_test.go
+++ b/internal/confgen/document_parser_test.go
@@ -2,13 +2,18 @@ package confgen
 
 import (
 	"context"
+	"fmt"
 	"testing"
 
 	"github.com/stretchr/testify/require"
 	"github.com/tableauio/tableau/format"
+	"github.com/tableauio/tableau/internal/importer"
 	"github.com/tableauio/tableau/internal/importer/book"
+	"github.com/tableauio/tableau/internal/importer/metasheet"
+	"github.com/tableauio/tableau/proto/tableaupb/internalpb"
 	"github.com/tableauio/tableau/proto/tableaupb/unittestpb"
 	"github.com/tableauio/tableau/xerrors"
+	"google.golang.org/protobuf/proto"
 )
 
 func newDocParserForTest() *sheetParser {
@@ -20,6 +25,74 @@ func newDocParserForTest() *sheetParser {
 			SubdirRewrites: map[string]string{},
 			BookFormat:     format.YAML,
 		})
+}
+
+func TestTableParser_parseDocumentMetasheet(t *testing.T) {
+	type args struct {
+		path string
+	}
+	tests := []struct {
+		name    string
+		parser  *sheetParser
+		args    args
+		wantErr bool
+		wantMsg proto.Message
+	}{
+		{
+			name:    "parse yaml metasheet",
+			parser:  newDocParserForTest(),
+			args:    args{path: "./testdata/Metasheet.yaml"},
+			wantErr: false,
+			wantMsg: &internalpb.Metabook{
+				MetasheetMap: map[string]*internalpb.Metasheet{
+					"HeroConf": {
+						Sheet: "HeroConf",
+					},
+					"ItemConf": {
+						Sheet:      "ItemConf",
+						Alias:      "ItemAliasConf",
+						OrderedMap: true,
+					},
+				},
+			},
+		},
+		{
+			name:    "parse xml metasheet",
+			parser:  newDocParserForTest(),
+			args:    args{path: "./testdata/Metasheet.xml"},
+			wantErr: false,
+			wantMsg: &internalpb.Metabook{
+				MetasheetMap: map[string]*internalpb.Metasheet{
+					"ItemConf": {
+						Sheet:      "ItemConf",
+						OrderedMap: true,
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			imp, err := importer.New(context.Background(), tt.args.path, importer.Parser(tt.parser))
+			if err != nil {
+				t.Fatal(err)
+			}
+			sheet := imp.GetSheet(metasheet.DefaultMetasheetName)
+			if sheet == nil {
+				t.Fatalf("metasheet not found")
+			}
+			msg := &internalpb.Metabook{}
+			err = tt.parser.Parse(msg, sheet)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("sheetParser.Parse() error = %s, wantErr %v", xerrors.NewDesc(err), tt.wantErr)
+			}
+			fmt.Println("sheet:", sheet)
+			fmt.Println("metabook:", msg)
+			if !proto.Equal(msg, tt.wantMsg) {
+				t.Errorf("\ngotMsg: %v\nwantMsg: %v", msg, tt.wantMsg)
+			}
+		})
+	}
 }
 
 func TestDocParser_parseFieldNotFound(t *testing.T) {

--- a/internal/confgen/fieldprop/refer.go
+++ b/internal/confgen/fieldprop/refer.go
@@ -158,13 +158,13 @@ func loadValueSpace(ctx context.Context, refer string, input *Input) (*ValueSpac
 	absWbPath := filepath.Join(input.InputDir, rewrittenWorkbookName)
 	primaryImporter, err := importer.New(ctx, absWbPath, importer.Sheets([]string{sheetName}))
 	if err != nil {
-		return nil, xerrors.WrapKV(err, xerrors.KeyModule, xerrors.ModuleConf, xerrors.KeyBookName, bookName)
+		return nil, xerrors.WrapKV(err, xerrors.KeyBookName, bookName)
 	}
 
 	// get merger importer infos
 	impInfos, err := importer.GetMergerImporters(ctx, input.InputDir, rewrittenWorkbookName, sheetName, sheetOpts.Merger, input.SubdirRewrites)
 	if err != nil {
-		return nil, xerrors.WrapKV(err, xerrors.KeyModule, xerrors.ModuleConf, xerrors.KeyBookName, bookName)
+		return nil, xerrors.WrapKV(err, xerrors.KeyBookName, bookName)
 	}
 
 	// append self

--- a/internal/confgen/parser.go
+++ b/internal/confgen/parser.go
@@ -109,7 +109,7 @@ func (x *sheetExporter) MergeAndExport(info *SheetInfo,
 	allImpInfos := append(impInfos, importer.ImporterInfo{Importer: mainImpInfo})
 	protomsg, err := ParseMessage(info, allImpInfos...)
 	if err != nil {
-		return err
+		return xerrors.WrapKV(err, xerrors.KeyModule, xerrors.ModuleConf)
 	}
 	// exported conf name pattern is : [ParentDir/]<SheetName>
 	getExportedConfName := func(info *SheetInfo, impInfo importer.ImporterInfo) string {
@@ -137,14 +137,13 @@ type oneMsg struct {
 func ParseMessage(info *SheetInfo, impInfos ...importer.ImporterInfo) (proto.Message, error) {
 	if len(impInfos) == 0 {
 		return nil, xerrors.ErrorKV("no importer to be parsed",
-			xerrors.KeyModule, xerrors.ModuleConf,
 			xerrors.KeyPrimaryBookName, info.PrimaryBookName,
 			xerrors.KeySheetName, info.SheetOpts.Name,
 			xerrors.KeyPBMessage, string(info.MD.Name()))
 	} else if len(impInfos) == 1 {
 		protomsg, err := parseMessageFromOneImporter(info, impInfos[0])
 		if err != nil {
-			return nil, xerrors.WrapKV(err, xerrors.KeyModule, xerrors.ModuleConf)
+			return nil, err
 		}
 		return protomsg, nil
 	}
@@ -174,7 +173,6 @@ func ParseMessage(info *SheetInfo, impInfos ...importer.ImporterInfo) (proto.Mes
 	}
 	if err := eg.Wait(); err != nil {
 		return nil, xerrors.WrapKV(err,
-			xerrors.KeyModule, xerrors.ModuleConf,
 			xerrors.KeyPrimaryBookName, info.PrimaryBookName,
 			xerrors.KeyPrimarySheetName, info.SheetOpts.Name)
 	}
@@ -194,7 +192,6 @@ func ParseMessage(info *SheetInfo, impInfos ...importer.ImporterInfo) (proto.Mes
 						bookNames := prevMsg.bookName + ", " + msg.bookName
 						sheetNames := prevMsg.sheetName + ", " + msg.sheetName
 						return nil, xerrors.WrapKV(err,
-							xerrors.KeyModule, xerrors.ModuleConf,
 							xerrors.KeyBookName, bookNames,
 							xerrors.KeySheetName, sheetNames,
 							xerrors.KeyPBMessage, string(info.MD.Name()))
@@ -202,7 +199,6 @@ func ParseMessage(info *SheetInfo, impInfos ...importer.ImporterInfo) (proto.Mes
 				}
 			}
 			return nil, xerrors.WrapKV(err,
-				xerrors.KeyModule, xerrors.ModuleConf,
 				xerrors.KeyBookName, msg.bookName,
 				xerrors.KeySheetName, msg.sheetName,
 				xerrors.KeyPBMessage, string(info.MD.Name()))
@@ -367,8 +363,13 @@ func (p *sheetParser) IsFieldOptional(field *Field) bool {
 	return p.sheetOpts.GetOptional() || field.opts.GetProp().GetOptional()
 }
 
-func (p *sheetParser) Parse(protomsg proto.Message, sheet *book.Sheet) error {
-	defer p.reset()
+func (p *sheetParser) Parse(protomsg proto.Message, sheet *book.Sheet) (err error) {
+	defer func() {
+		p.reset()
+		if err != nil {
+			err = xerrors.WrapKV(err, xerrors.KeyModule, xerrors.ModuleConf)
+		}
+	}()
 	if sheet.Document != nil {
 		docParser := &documentParser{sheetParser: p}
 		return docParser.Parse(protomsg, sheet)

--- a/internal/confgen/table_parser_test.go
+++ b/internal/confgen/table_parser_test.go
@@ -2,15 +2,12 @@ package confgen
 
 import (
 	"context"
-	"fmt"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"github.com/tableauio/tableau/format"
-	"github.com/tableauio/tableau/internal/importer"
 	"github.com/tableauio/tableau/internal/importer/book"
-	"github.com/tableauio/tableau/internal/importer/metasheet"
 	"github.com/tableauio/tableau/proto/tableaupb"
 	"github.com/tableauio/tableau/proto/tableaupb/internalpb"
 	"github.com/tableauio/tableau/proto/tableaupb/unittestpb"
@@ -27,6 +24,48 @@ func newTableParserForTest() *sheetParser {
 			SubdirRewrites: map[string]string{},
 			BookFormat:     format.CSV,
 		})
+}
+
+func TestTableParser_parseTableMetasheet(t *testing.T) {
+	type args struct {
+		sheet *book.Sheet
+	}
+	tests := []struct {
+		name    string
+		parser  *sheetParser
+		args    args
+		wantErr bool
+		err     error
+	}{
+		{
+			name:   "no empty row",
+			parser: newTableParserForTest(),
+			args: args{
+				sheet: book.NewTableSheet(
+					"@TABLEAU",
+					[][]string{
+						{"Sheet", "OrderedMap", "Mode"},
+						{"Item", "", ""},
+						{"Enum", "MODE_ENUM_TYPE", ""},
+					}),
+			},
+			wantErr: true,
+			err:     xerrors.ErrE2013,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := tt.parser.Parse(&internalpb.Metasheet{}, tt.args.sheet)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("sheetParser.Parse() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if err != nil {
+				// t.Logf("err: %v", xerrors.NewDesc(err))
+				require.Equal(t, xerrors.ModuleConf, xerrors.NewDesc(err).GetValue(xerrors.KeyModule))
+				require.ErrorIs(t, err, tt.err)
+			}
+		})
+	}
 }
 
 func TestTableParser_parseVerticalMapWithDuplicateKey(t *testing.T) {
@@ -419,88 +458,6 @@ func TestTableParser_parseHorizontalMapWithEmptyKey(t *testing.T) {
 			}
 			if err != nil {
 				require.ErrorIs(t, err, tt.err)
-			}
-		})
-	}
-}
-
-func TestTableParser_parseDocumentMetasheet(t *testing.T) {
-	type args struct {
-		path string
-	}
-	tests := []struct {
-		name    string
-		parser  *sheetParser
-		args    args
-		wantErr bool
-		wantMsg proto.Message
-	}{
-		{
-			name: "parse yaml metasheet",
-			parser: NewExtendedSheetParser(context.Background(), "protoconf", "Asia/Shanghai",
-				book.MetabookOptions(),
-				book.MetasheetOptions(context.Background()),
-				&SheetParserExtInfo{
-					InputDir:       "",
-					SubdirRewrites: map[string]string{},
-					BookFormat:     format.YAML,
-				}),
-			args:    args{path: "./testdata/Metasheet.yaml"},
-			wantErr: false,
-			wantMsg: &internalpb.Metabook{
-				MetasheetMap: map[string]*internalpb.Metasheet{
-					"HeroConf": {
-						Sheet: "HeroConf",
-					},
-					"ItemConf": {
-						Sheet:      "ItemConf",
-						Alias:      "ItemAliasConf",
-						OrderedMap: true,
-					},
-				},
-			},
-		},
-		{
-			name: "parse xml metasheet",
-			parser: NewExtendedSheetParser(context.Background(), "protoconf", "Asia/Shanghai",
-				book.MetabookOptions(),
-				book.MetasheetOptions(context.Background()),
-				&SheetParserExtInfo{
-					InputDir:       "",
-					SubdirRewrites: map[string]string{},
-					BookFormat:     format.XML,
-				}),
-			args:    args{path: "./testdata/Metasheet.xml"},
-			wantErr: false,
-			wantMsg: &internalpb.Metabook{
-				MetasheetMap: map[string]*internalpb.Metasheet{
-					"ItemConf": {
-						Sheet:      "ItemConf",
-						OrderedMap: true,
-					},
-				},
-			},
-		},
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			imp, err := importer.New(context.Background(), tt.args.path, importer.Parser(tt.parser))
-			if err != nil {
-				t.Fatal(err)
-			}
-			sheet := imp.GetSheet(metasheet.DefaultMetasheetName)
-			if sheet == nil {
-				t.Fatalf("metasheet not found")
-			}
-			msg := &internalpb.Metabook{}
-			err = tt.parser.Parse(msg, sheet)
-			if (err != nil) != tt.wantErr {
-				t.Errorf("sheetParser.Parse() error = %s, wantErr %v", xerrors.NewDesc(err), tt.wantErr)
-			}
-			fmt.Println("sheet:", sheet)
-			fmt.Println("metabook:", msg)
-			if !proto.Equal(msg, tt.wantMsg) {
-				t.Errorf("\ngotMsg: %v\nwantMsg: %v", msg, tt.wantMsg)
 			}
 		})
 	}

